### PR TITLE
[CORDA-2243] Notary client should sign over NotarisationRequest serialised with object referencing turned off

### PR DESCRIPTION
**Pull request by thschroeter**
To make sure the notary service constructs the same serialised bytes as the client.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryUtils.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ fun NotarisationResponse.validateSignatures(txId: SecureHash, notary: Party) {
 }
 
 /** Creates a signature over the notarisation request using the legal identity key. */
-fun NotarisationRequest.generateSignature(serviceHub: ServiceHub): NotarisationRequestSignature {
+fun NotarisationRequest.generateRequestSignature(serviceHub: ServiceHub): NotarisationRequestSignature {
```
Comment (adagys) [inline core/src/main/kotlin/net/corda/core/internal/notary/NotaryUtils.kt]: was this rename intended? Seems as if intellij was trying to be too clever!

Comment (thschroeter) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryUtils.kt]: right, i'll undo it
